broadinstitute/scp-ingest-pipeline
# Enable merging cell types from CAS into AnnData via CLI (SCP-5732)

This supports adding ontology-aware [CAS](https://www.cellarium.ai/tool/cellarium-cell-annotation-service-cas/) cell type annotations into an AnnData object that can initialize an SCP study.

Here's how the resulting study looks!

https://github.com/user-attachments/assets/d8978879-661e-44a7-8399-23534bd6d91c

[Previously](https://github.com/broadinstitute/scp-ingest-pipeline/pull/353), we could call CAS for an AnnData file and print results to the terminal.  Now, we can integrate CAS results into the AnnData file in a manner that can be usefully searched and interactively visualized in a study.  

The full CAS results JSON is also saved, as are interstitial copies.  The CAS JSON seems useful to store in a bucket.  The interstitial AnnData is mostly a helpful diagnostics tool that we can discard as this matures, if we desire.

### Test
1.  Do steps from [PR #353](https://github.com/broadinstitute/scp-ingest-pipeline/pull/353)
2. Confirm output file exists: `cas_annotated_trimmed_compliant__pbmc_10x_v3_4k.h5ad`
3. Create a study
4. Upload `cas_annotated_trimmed_compliant__pbmc_10x_v3_4k.h5ad` as an AnnData file
5. Confirm annotations like "cell_type_cas_1", "cell_type_cas_2", and "cell_type_cas_3" appear
6. Confirm search and interactive visualizations work as expected

This satisfies SCP-5732.
